federation/pkg/federation-controller/secret/secret_controller.go

```diff
@@ -175,20 +176,32 @@ func (secretcontroller *SecretController) Run(stopChan <-chan struct{}) {
 	}()
 }
 
+func getSecretKey(namespace, name string) string {
+	return fmt.Sprintf("%s/%s", namespace, name)
+}
+
+// Internal structure for data in delaying deliverer.
+type secretItem struct {
```

Comment:
I would suggest that you rather use the existing types.NamespacedName here.  But that can go in a followup PR if you prefer.